Added some failing integral tests

oscargus opened the pull request:
<!-- Your title above should be a short description of what
was changed. Do not include the issue number in the title. -->

#### References to other Issues or PRs
<!-- If this pull request fixes an issue, write "Fixes #NNNN" in that exact
format, e.g. "Fixes #1234". See
https://github.com/blog/1506-closing-issues-via-pull-requests . Please also
write a comment on that issue linking back to this pull request once it is
open. -->


#### Brief description of what is fixed or changed
Added failing integral examples from #14074, #14078, #14398, #14709, #14716, #15227, #15925, #16084, #16161,  #17058

More examples in later commit: #7109, #7147, #7264, #7161, #9101, #9723, #10584, #11254, #11742, #11813, #11845

Also added passing integral example from #11876, #14078

#### Other comments
The idea is clearly that once any of these tests does pass, the corresponding issue may be closed.

#### Release Notes

<!-- Write the release notes for this release below. See
https://github.com/sympy/sympy/wiki/Writing-Release-Notes for more information
on how to write release notes. The bot will check your release notes
automatically to see if they are formatted correctly. -->

<!-- BEGIN RELEASE NOTES -->
NO ENTRY
<!-- END RELEASE NOTES -->

oscargus committed:
Added some failing integral tests

oscargus commented:
Just realized the pretty obvious thing that it is faster to evaluate it only once and use the result for both assertions... Will update.

smichr commented on sympy/integrals/tests/test_failing_integrals.py:
```diff
@@ -114,3 +114,79 @@ def test_issue_4941():
+
+@XFAIL
+def test_issue_16084():
+    assert not integrate(log(sin(x)), (x, 0, pi/2)).has(Integral)
```
I'm not sure this first assertion is necessary. But do you have it there in case the integral is solved but in a different form?

oscargus replied:
I think the first thing is to be able to solve the integral in any form. For the cases where a compact answer is known, there is a second assert for that answer. It can of course be argued if the issue is closed by returning a complicated expression or not, but that is my logic behind it.

See e.g. 
https://github.com/sympy/sympy/blob/d12570e481d862f8e06292e9684b97bc7f5b4821/sympy/integrals/tests/test_failing_integrals.py#L35-L40

Maybe one should actually have two different tests, one for solving it and one for solving it "correctly"? Browsing through the failing integral tests made me realize that many of the issues had been closed, but the tests were not removed from `test_failing_integrals`. I guess the reason is that someone solves a general problem and then it may have positive effects on other parts of the code as well (although that doesn't really explain why the issues were closed...).

oscargus committed:
Some more tests and a rewrite

oscargus commented:
Maybe one should write a dedicated function here that takes the expression, integration variables, and possibly known result and prints out a bit more details in case one of these integrals "suddenly" starts to work?

oscargus commented:
I will disable the 595 seconds test...

asmeurer commented:
Yes, a lot of the XFAIL tests aren't written in a good way, because some mistake in the test could mean they will still fail even if the functionality starts working.

asmeurer commented:
Especially here they test that it isn't an unevaluated integral but then it also tests for the answer. But if the answer comes out differently, it won't pass. That is actually quite likely for integrals, because different algorithms can produce different answers, sometimes just formatted differently, sometimes mathematically different (because indefinite integrals are only equal up to a constant).

smichr commented on sympy/integrals/tests/test_failing_integrals.py:
```diff
@@ -114,3 +114,154 @@ def test_issue_4941():
+def test_issue_16084():
+    i = integrate(log(sin(x)), (x, 0, pi/2))
+    assert not i.has(Integral)
+    assert i == -pi*log(2)/2
```
```suggestion
    # assert i == -pi*log(2)/2
```

smichr commented on sympy/integrals/tests/test_failing_integrals.py:
```diff
@@ -114,3 +114,154 @@ def test_issue_4941():
+def test_issue_16161():
+    i = integrate(x*sec(x)**2, x)
+    assert not i.has(Integral)
+    assert i == x*tan(x) + log(cos(x))
```
```suggestion
    # assert i == x*tan(x) + log(cos(x))
```

smichr commented on sympy/integrals/tests/test_failing_integrals.py:
```diff
@@ -114,3 +114,154 @@ def test_issue_4941():
+        skip("Too slow for travis.")
+    i = integrate(log(1-x)*log((1+x)**2)/x, (x, 0, 1))
+    assert not i.has(Integral)
+    assert i == -5*zeta(3)/4
```
```suggestion
    # assert i == -5*zeta(3)/4
```

smichr commented on sympy/integrals/tests/test_failing_integrals.py:
```diff
@@ -114,3 +114,154 @@ def test_issue_4941():
+    # Mathematica can not solve it either, but
+    # integrate(log(x + 5)*cos(pi*x),(x, S.Half, 1)).transform(x, y - 5).doit()
+    # works
+    assert i == -log(S(11)/2)/pi - Si(11*pi/2)/pi + Si(6*pi)/pi
```
```suggestion
    # assert i == -log(S(11)/2)/pi - Si(11*pi/2)/pi + Si(6*pi)/pi
```

smichr commented on sympy/integrals/tests/test_failing_integrals.py:
```diff
@@ -114,3 +114,154 @@ def test_issue_4941():
+def test_issue_14709a():
+    i = integrate(x*acos(1 - 2*x/h), (x, 0, h))
+    assert not i.has(Integral)
+    assert i == 5*h**2*pi/16
```
```suggestion
    # assert i == 5*h**2*pi/16
```

smichr commented on sympy/integrals/tests/test_failing_integrals.py:
```diff
@@ -114,3 +114,154 @@ def test_issue_4941():
+def test_issue_14074():
+    i = integrate(log(sin(x)), (x, 0, pi/2))
+    assert not i.has(Integral)
+    assert i == -pi*log(2)/2
```
```suggestion
    # assert i == -pi*log(2)/2
```

smichr commented on sympy/integrals/tests/test_failing_integrals.py:
```diff
@@ -114,3 +114,154 @@ def test_issue_4941():
+def test_issue_14078b():
+    i = integrate((atan(4*x)-atan(2*x))/x, (x, 0, oo))
+    assert not i.has(Integral)
+    assert i == pi*log(2)/2
```
```suggestion
    # assert i == pi*log(2)/2
```

smichr commented on sympy/integrals/tests/test_failing_integrals.py:
```diff
@@ -114,3 +114,154 @@ def test_issue_4941():
+def test_issue_11742():
+    i =  integrate(sqrt(-x**2 + 8*x + 48), (x, 4, 12))
+    assert not i.has(Integral)
+    assert i == 16*pi
```
```suggestion
    # assert i == 16*pi
```

smichr commented:
I commented out expected answer; they can be uncommented once a test passes.

smichr committed:
comment out expected answer

When the XFAIL passes the assertion that there is no Integral can be replaced with an assertion of the result (or equivalent).

oscargus commented:
Makes sense to comment out the expected answers. Thanks!

I was actually thinking if it made sense to run these tests every time? On one hand, it is good to keep track of old issues. On the other hand, they will probably not magically start working (unless someone actually works on integrals).  Not sure how it can be handled though.
